x/custom/msg.go

```diff
@@ -0,0 +1,32 @@
+package custom
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+)
+
+func init() {
+	migration.MustRegister(1, &CreateCustomStateMsg{}, migration.NoModification)
+}
+
+const (
+	pathCreateCustomStateMsg = "custom/create_custom_state"
+)
+
+var _ weave.Msg = (*CreateCustomStateMsg)(nil)
+
+func (CreateCustomStateMsg) Path() string {
+	return pathCreateCustomStateMsg
+}
+
+func (m CreateCustomStateMsg) Validate() error {
+	if err := m.Metadata.Validate(); err != nil {
```

Comment:
You can make this better by using a multi error. We are planning to [migrate all validation errors](https://github.com/iov-one/weave/issues/890) into using multi error. Required code is present for quite some time in weave now. Migration is not finished because we never took the time to update our validation code.
You can see the [example use in the `x/paychan` codebase](https://github.com/iov-one/weave/blob/master/x/paychan/model.go#L16-L49).

Reply:
Yes, please use the newest idioms. Thanks for adding this Piotr. 

We will need to update all docs and sample for the best practices we agreed on after some iterations the last few months.